### PR TITLE
Added documentation to the QR package

### LINKED ISSUE #187: Document the files in mifopay/qr package
**Summary:**

Document the files in mifopay/qr package: https://github.com/openMF/mobile-wallet/tree/master/mifospay/src/main/java/org/mifos/mobilewallet/mifospay/qr

### PR DESCRIPTION
**Pull request by codepoet2017390**
Fixes #187

Added documentation to qr package

Please make sure these boxes are checked before submitting your pull request - thanks!

- [x] Apply the `AndroidStyle.xml` style template to your code in Android Studio.

- [x] Run the unit tests with `./gradlew check` to make sure you didn't break anything

- [x] If you have multiple commits please combine them into one commit by squashing them.

### CODE REVIEW

**Comment by codepoet2017390** [comment]
@ankurs287 @luckyman20 please review

File: mifospay/src/main/java/org/mifos/mobilewallet/mifospay/qr/domain/usecase/GenerateQr.java [hunk 1]
```diff
@@ -26,6 +28,11 @@
     public GenerateQr() {
     }
 
+    /**
+     * method generates QR as a bitmap
```
Comment (luckyman20) [inline mifospay/src/main/java/org/mifos/mobilewallet/mifospay/qr/domain/usecase/GenerateQr.java]:
```suggestion
     * Method generates QR as a bitmap.
```

File: mifospay/src/main/java/org/mifos/mobilewallet/mifospay/qr/domain/usecase/GenerateQr.java [hunk 2]
```diff
@@ -26,6 +28,11 @@
...
+     * called when {@link org.mifos.mobilewallet.core.base.UseCaseScheduler} executes run()
```
Comment (luckyman20) [inline mifospay/src/main/java/org/mifos/mobilewallet/mifospay/qr/domain/usecase/GenerateQr.java]:
```suggestion
     * Called when {@link org.mifos.mobilewallet.core.base.UseCaseScheduler} executes run().
```

File: mifospay/src/main/java/org/mifos/mobilewallet/mifospay/qr/domain/usecase/GenerateQr.java [hunk 3]
```diff
@@ -43,6 +50,8 @@ protected void executeUseCase(RequestValues requestValues) {
         }
     }
 
+    //Encodes a bitmap from the given data and returns
+    //it to executeUseCase method
```
Comment (luckyman20) [inline mifospay/src/main/java/org/mifos/mobilewallet/mifospay/qr/domain/usecase/GenerateQr.java]: Use Javadoc method only. These type of comments are mostly used for TODOs.

**Comment by codepoet2017390** [comment]
@luckyman20  updated

**Comment by luckyman20** [comment]
Use Javadoc comments everywhere.

**Comment by codepoet2017390** [comment]
@luckyman20  updated

File: mifospay/src/main/java/org/mifos/mobilewallet/mifospay/qr/domain/usecase/GenerateQr.java [hunk 4]
```diff
@@ -65,6 +77,9 @@ private Bitmap encodeAsBitmap(String str) throws WriterException {
         return bitmap;
     }
 
+    /**
+     * Data passed to a request
```
Comment (luckyman20) [inline mifospay/src/main/java/org/mifos/mobilewallet/mifospay/qr/domain/usecase/GenerateQr.java]: Please specify what kind of data is passed.

File: mifospay/src/main/java/org/mifos/mobilewallet/mifospay/qr/presenter/ReadQrPresenter.java [hunk 5]
```diff
@@ -16,11 +18,15 @@
     private QrContract.ReadQrView mReadQrView;
 
 
+
```
Comment (luckyman20) [inline mifospay/src/main/java/org/mifos/mobilewallet/mifospay/qr/presenter/ReadQrPresenter.java]:
```suggestion
```

File: mifospay/src/main/java/org/mifos/mobilewallet/mifospay/qr/presenter/ShowQrPresenter.java [hunk 6]
```diff
@@ -24,13 +25,20 @@ public ShowQrPresenter(UseCaseHandler useCaseHandler) {
         this.mUsecaseHandler = useCaseHandler;
     }
 
+    /**
+     * method to attach view to presenter
```
Comment (luckyman20) [inline mifospay/src/main/java/org/mifos/mobilewallet/mifospay/qr/presenter/ShowQrPresenter.java]:
```suggestion
     * Method to attach view to presenter.
```

File: mifospay/src/main/java/org/mifos/mobilewallet/mifospay/qr/ui/ReadQrActivity.java [hunk 7]
```diff
@@ -73,6 +73,9 @@ public void handleResult(Result result) {
 
     }
 
+    /**
+     *set the presenter to this view
```
Comment (luckyman20) [inline mifospay/src/main/java/org/mifos/mobilewallet/mifospay/qr/ui/ReadQrActivity.java]:
```suggestion
     * Set the presenter to this view.
```

File: mifospay/src/main/java/org/mifos/mobilewallet/mifospay/qr/ui/ShowQrActivity.java [hunk 8]
```diff
@@ -53,16 +53,23 @@ protected void onCreate(@Nullable Bundle savedInstanceState) {
         mShowQrPresenter.generateQr(qrData);
         tvQrData.setText(qrData);
 
+        //raising screen brightness to full
```
Comment (luckyman20) [inline mifospay/src/main/java/org/mifos/mobilewallet/mifospay/qr/ui/ShowQrActivity.java]: You left this.

**Comment by codepoet2017390** [comment]
@luckyman20  updated

File: mifospay/src/main/java/org/mifos/mobilewallet/mifospay/qr/domain/usecase/GenerateQr.java [hunk 9]
```diff
@@ -43,6 +50,11 @@ protected void executeUseCase(RequestValues requestValues) {
         }
     }
 
+    /**
+     * Encodes a bitmap from the given data and returns
+     * it to executeUseCase method
+     * @param str vpa of the user
```
Comment (luckyman20) [inline mifospay/src/main/java/org/mifos/mobilewallet/mifospay/qr/domain/usecase/GenerateQr.java]:
```suggestion
     * @param str VPA of the user
```

File: mifospay/src/main/java/org/mifos/mobilewallet/mifospay/qr/domain/usecase/GenerateQr.java [hunk 10]
```diff
@@ -65,6 +77,9 @@ private Bitmap encodeAsBitmap(String str) throws WriterException {
         return bitmap;
     }
 
+    /**
+     * Data containing vpa of user passed to a request
```
Comment (luckyman20) [inline mifospay/src/main/java/org/mifos/mobilewallet/mifospay/qr/domain/usecase/GenerateQr.java]:
```suggestion
     * Data containing VPA of user passed to a request
```

File: mifospay/src/main/java/org/mifos/mobilewallet/mifospay/qr/domain/usecase/GenerateQr.java [hunk 11]
```diff
@@ -74,6 +89,9 @@ public RequestValues(String data) {
         }
     }
 
+    /**
+     * Data(final QR bitmap) received from requests
```
Comment (luckyman20) [inline mifospay/src/main/java/org/mifos/mobilewallet/mifospay/qr/domain/usecase/GenerateQr.java]:
```suggestion
     * Data (final QR bitmap) received from requests
```

File: mifospay/src/main/java/org/mifos/mobilewallet/mifospay/qr/presenter/ReadQrPresenter.java [hunk 12]
```diff
@@ -7,20 +7,25 @@
...
 
 public class ReadQrPresenter implements QrContract.ReadQrPresenter {
 
     private final UseCaseHandler mUsecaseHandler;
     private QrContract.ReadQrView mReadQrView;
 
-
+    
     @Inject
     public ReadQrPresenter(UseCaseHandler useCaseHandler) {
```
Comment (luckyman20) [inline mifospay/src/main/java/org/mifos/mobilewallet/mifospay/qr/presenter/ReadQrPresenter.java]: Document this as well.

File: mifospay/src/main/java/org/mifos/mobilewallet/mifospay/qr/ui/ReadQrActivity.java [hunk 13]
```diff
@@ -73,6 +73,9 @@ public void handleResult(Result result) {
 
     }
 
+    /**
+     *Set the presenter to this view
```
Comment (luckyman20) [inline mifospay/src/main/java/org/mifos/mobilewallet/mifospay/qr/ui/ReadQrActivity.java]:
```suggestion
     * Set the presenter to this view
```

File: mifospay/src/main/java/org/mifos/mobilewallet/mifospay/qr/presenter/ReadQrPresenter.java (hunk 12 above)
Comment (codepoet2017390) [reply mifospay/src/main/java/org/mifos/mobilewallet/mifospay/qr/presenter/ReadQrPresenter.java]: This is just the default constructor does not need any explanation

Comment (luckyman20) [reply mifospay/src/main/java/org/mifos/mobilewallet/mifospay/qr/presenter/ReadQrPresenter.java]: You can simply put `Initializes ReadQrPresenter Class`

File: mifospay/src/main/java/org/mifos/mobilewallet/mifospay/qr/presenter/ShowQrPresenter.java [hunk 14]
```diff
@@ -24,13 +25,20 @@ public ShowQrPresenter(UseCaseHandler useCaseHandler) {
...
     @Override
     public void attachView(BaseView baseView) {
         this.mShowQrView = (QrContract.ShowQrView) baseView;
         mShowQrView.setPresenter(this);
 
     }
 
+    /**
+     * Method used to generate QR code and display it to the user
+     * @param data vpa of the user
```
Comment (luckyman20) [inline mifospay/src/main/java/org/mifos/mobilewallet/mifospay/qr/presenter/ShowQrPresenter.java]:
```suggestion
     * @param data VPA of the user
```

File: mifospay/src/main/java/org/mifos/mobilewallet/mifospay/qr/domain/usecase/GenerateQr.java [hunk 15]
```diff
@@ -13,7 +13,9 @@
 import javax.inject.Inject;
 
 /**
- * Created by naman on 8/7/17.
+ * GenerateQr class generates QR codes for users when ShowQR is clicked
```
Comment (luckyman20) [inline mifospay/src/main/java/org/mifos/mobilewallet/mifospay/qr/domain/usecase/GenerateQr.java]:
```suggestion
 * GenerateQr class which generates QR codes for users when ShowQR is clicked.
```

File: mifospay/src/main/java/org/mifos/mobilewallet/mifospay/qr/domain/usecase/GenerateQr.java [hunk 16]
```diff
@@ -26,6 +28,11 @@
     public GenerateQr() {
     }
 
+    /**
+     * Method generates QR as a bitmap.
```
Comment (luckyman20) [inline mifospay/src/main/java/org/mifos/mobilewallet/mifospay/qr/domain/usecase/GenerateQr.java]:
```suggestion
     * This method generates QR as a bitmap.
```

File: mifospay/src/main/java/org/mifos/mobilewallet/mifospay/qr/domain/usecase/GenerateQr.java [hunk 17]
```diff
@@ -43,6 +50,11 @@ protected void executeUseCase(RequestValues requestValues) {
         }
     }
 
+    /**
+     * Encodes a bitmap from the given data and returns
+     * it to executeUseCase method
```
Comment (luckyman20) [inline mifospay/src/main/java/org/mifos/mobilewallet/mifospay/qr/domain/usecase/GenerateQr.java]: Keep it in one line only.

File: mifospay/src/main/java/org/mifos/mobilewallet/mifospay/qr/presenter/ReadQrPresenter.java [hunk 18]
```diff
@@ -7,20 +7,28 @@
...
 
 public class ReadQrPresenter implements QrContract.ReadQrPresenter {
 
     private final UseCaseHandler mUsecaseHandler;
     private QrContract.ReadQrView mReadQrView;
 
-
+    /**
+     * Initializes the ReadQrPresenter class
+     * @param useCaseHandler instance of UseCaseHandler class
```
Comment (luckyman20) [inline mifospay/src/main/java/org/mifos/mobilewallet/mifospay/qr/presenter/ReadQrPresenter.java]:
```suggestion
     * @param useCaseHandler Instance of UseCaseHandler class.
```

File: mifospay/src/main/java/org/mifos/mobilewallet/mifospay/qr/ui/ShowQrActivity.java [hunk 19]
```diff
@@ -58,11 +58,17 @@ protected void onCreate(@Nullable Bundle savedInstanceState) {
...
+    /**
+     * Set presenter to this view
+     */
     @Override
     public void setPresenter(QrContract.ShowQrPresenter presenter) {
         this.mShowQrPresenter = presenter;
     }
 
+    /**
+     * Display the generated QR in the image view
```
Comment (luckyman20) [inline mifospay/src/main/java/org/mifos/mobilewallet/mifospay/qr/ui/ShowQrActivity.java]:
```suggestion
     * Displays the generated QR in the image view.
```

**Comment by luckyman20** [comment]
Also, squash your commits.

**Comment by luckyman20** [comment]
@codepoet2017390 Can you please do the required changes so that we can go ahead with merging this?

**Comment by luckyman20** [comment]
@codepoet2017390 We will wait for another 3 days and if you don't reply we have to close this PR.

**Comment by codepoet2017390** [comment]
Hi @luckyman20, sorry for being completely inactive lately. I have my exams going on. Will do the required changes as soon as I get time.

**Comment by droidchef** [comment]
Please reopen this PR after you get a chance to address the comments from the reviewers.

**Pull request closed without merging**